payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayLauncherViewModel.kt

```diff
@@ -156,15 +154,14 @@ internal class GooglePayLauncherViewModel(
         }
     }
 
-    suspend fun createLoadPaymentDataTask(): Task<PaymentData> {
-        check(isReadyToPay()) {
-            "Google Pay is unavailable."
+    suspend fun createLoadPaymentDataTask(): Result<Task<PaymentData>> {
+        if (!isReadyToPay()) {
+            return Result.failure(IllegalStateException("Google Pay is unavailable."))
         }
```

Comment:
Returning `Result.failure()` instead of throwing because this method is no longer wrapped in a `runCatching`.